salt/modules/aixpkg.py

```diff
@@ -350,31 +558,91 @@ def remove(name=None, pkgs=None, **kwargs):
 
 def latest_version(*names, **kwargs):
     """
-    Return the latest version of the named fileset/rpm package available for
+    Return the latest available version of the named fileset/rpm package available for
     upgrade or installation. If more than one fileset/rpm package name is
     specified, a dict of name/version pairs is returned.
 
     If the latest version of a given fileset/rpm package is already installed,
     an empty string will be returned for that package.
 
+    .. versionchanged:: 3005
+
     CLI Example:
 
     .. code-block:: bash
 
         salt '*' pkg.latest_version <package name>
         salt '*' pkg.latest_version <package1> <package2> <package3> ...
 
-    NOTE: Repositories are not presently supported for AIX.
-    This function will always return an empty string for a given
-    fileset/rpm package.
+    Note: currently only functional for rpm packages due to filesets do not have a specific location to check
+        Requires yum of dnf available in order to query a repository
+
+    This function will always return an empty string for unfound fileset/rpm package.
     """
     kwargs.pop("refresh", True)
 
     ret = {}
     if not names:
         return ""
     for name in names:
-        ret[name] = ""
+        # AIX packaging includes info on filesets and rpms
+        version_found = ""
+        libpathenv = {"LIBPATH": "/opt/freeware/lib:/usr/lib"}
+        if pathlib.Path("/opt/freeware/bin/dnf").is_file():
+            cmdexe = "/opt/freeware/bin/dnf"
+            cmd = f"{cmdexe} check-update {name}"
+            available_info = __salt__["cmd.run_all"](
+                cmd, python_shell=False, env=libpathenv, ignore_retcode=True
+            )
+        elif pathlib.Path("/opt/freeware/bin/yum").is_file():
+            cmdexe = "/opt/freeware/bin/yum"
+            cmd = f"{cmdexe} check-update {name}"
+            available_info = __salt__["cmd.run_all"](
+                cmd, python_shell=False, env=libpathenv, ignore_retcode=True
+            )
+        elif pathlib.Path("/usr/bin/yum").is_file():
+            cmdexe = "/usr/bin/yum"
+            cmd = f"{cmdexe} check-update {name}"
+            available_info = __salt__["cmd.run_all"](
+                cmd, python_shell=False, env=libpathenv, ignore_retcode=True
+            )
+        else:
+            # no yum found implies no repository support
+            available_info = None
+
+        log.debug(
+            f"latest_version dnf|yum check-update command '{cmd}', available_info is '{available_info}'"
+        )
+        if available_info and (
+            0 == available_info["retcode"] or 100 == available_info["retcode"]
+        ):
+            available_output = available_info["stdout"]
+            if available_output:
+                available_list = available_output.split()
+                log.debug(
+                    f"available_list is length '{len(available_list)}', list '{available_list}'"
```

Comment:
no f strings and more useful log for end user.

Reply:
removed message